libs/platform/src/lib/components/form/checkbox-group/checkbox-group.component.spec.ts

```diff
@@ -0,0 +1,451 @@
+import { FormModule } from '@fundamental-ngx/core';
+import { FdpFormGroupModule } from './../form-group/fdp-form.module';
+import { CheckboxGroupComponent } from './checkbox-group.component';
+import { CheckboxComponent } from '../checkbox/checkbox.component';
+import { SelectItem } from '../../../domain/data-model';
+import { PlatformCheckboxModule } from '../checkbox/checkbox.module';
+import { By } from '@angular/platform-browser';
+
+@Component({
+    selector: 'fdp-cbg-reactive-test',
+    template: `
+        <fdp-form-group [multiLayout]="true" [formGroup]="form1" [object]="formData">
+            <fdp-form-field #fl1 [id]="'phones'" [label]="'Phones interested in:'" zone="zLeft" rank="1">
+                <fdp-checkbox-group
+                    [list]="phoneslist"
+                    [name]="'brands'"
+                    [formControl]="fl1.formControl"
+                ></fdp-checkbox-group>
+            </fdp-form-field>
+
+            <fdp-form-field #fl2 [id]="'visited'" [label]="'Country visited: '" zone="zLeft" rank="1">
+                <fdp-checkbox-group
+                    [list]="countryVisited"
+                    [name]="'visited'"
+                    [formControl]="fl2.formControl"
+                ></fdp-checkbox-group>
+            </fdp-form-field>
+
+            <fdp-form-field #fl3 [id]="'hobbies'" [label]="'My Hobbies:'" zone="zLeft" rank="1">
+                <fdp-checkbox-group [name]="'hobby'" [formControl]="fl3.formControl">
+                    <fdp-checkbox [value]="'cooking'" [label]="'Cooking'"></fdp-checkbox>
+                    <fdp-checkbox [value]="'painting'" [label]="'Painting'"></fdp-checkbox>
+                    <fdp-checkbox [value]="'coding'" [label]="'Coding'"></fdp-checkbox>
+                    <fdp-checkbox [value]="'gardening'" [label]="'Gardening'"></fdp-checkbox>
+                </fdp-checkbox-group>
+            </fdp-form-field>
+
+            <fdp-form-field #fl4 [id]="'language'" [label]="'Languages Known: '" zone="zLeft" rank="1">
+                <fdp-checkbox-group
+                    [list]="languages"
+                    [name]="'language'"
+                    [formControl]="fl4.formControl"
+                ></fdp-checkbox-group>
+            </fdp-form-field>
+
+            <fdp-form-field #fl5 [id]="'fruits'" [label]="'Fruits:'" zone="zLeft" rank="1">
+                <fdp-checkbox-group [name]="'fruits'" [formControl]="fl5.formControl">
+                    <fdp-checkbox [value]="'apple'" [label]="'Apple'" [disabled]="true"></fdp-checkbox>
+                    <fdp-checkbox [value]="'banana'" [label]="'Banana'"></fdp-checkbox>
+                    <fdp-checkbox [value]="'guava'" [label]="'Guava'" [disabled]="true"></fdp-checkbox>
+                    <fdp-checkbox [value]="'papaya'" [label]="'Papaya'"></fdp-checkbox>
+                </fdp-checkbox-group>
+            </fdp-form-field>
+        </fdp-form-group>
+    `
+})
+class TestReactiveCheckboxGroupComponnet {
```

Comment:
Also create test to use displayKey and Lookup Key

Reply:
included tests and examples using displayKey and Lookup Key